### PR TITLE
Gps breakout

### PR DESCRIPTION
**Pull request by VivaanBahl**
Breaks the code out into more testable sections

### CODE REVIEW

File: Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp [hunk 1]
```diff
@@ -41,87 +38,118 @@ int Transistor_GPS_Broadcaster::handle_serial_messages()
...
+        ROS_ERROR_STREAM("read a GPGGA when it wasn't!");
+        return false;
+    }
+    else if (!gps_locked(tokens))
+    {
+        ROS_ERROR_STREAM("Not locked yet!");
+    }
+    else
+    {
+        double latitude_deg = 41.414938;//convert_to_latitude(tokens[2]);
```
Comment (VivaanBahl) [inline Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp]: should be converted back into the original code

File: Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp [hunk 2]
```diff
@@ -41,87 +38,118 @@ int Transistor_GPS_Broadcaster::handle_serial_messages()
...
+        double longitude_deg = 81.86139667;//convert_to_longitude(tokens[4]);
+
+        //if(tokens[5] == "W") longitude_deg *= -1;
```
Comment (VivaanBahl) [inline Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp]: uncomment

File: Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp [hunk 3]
```diff
@@ -41,87 +38,118 @@ int Transistor_GPS_Broadcaster::handle_serial_messages()
...
+    gps_point.longitude = longitude_deg;
+
+    geodesy::UTMPoint utm_point(gps_point);
+
+    return utm_point;
+}
+
+robobuggy::GPS Transistor_GPS_Broadcaster::gps_gen_message(double latitude_deg, double longitude_deg, geodesy::UTMPoint utm_point) {
+    robobuggy::GPS gps_message;
+    gps_message.Lat_deg = static_cast<float>(latitude_deg);
```
Comment (VivaanBahl) [inline Software/real_time/ROS_RoboBuggy/src/robobuggy/src/gps/Transistor_GPS_Broadcaster.cpp]: is the `static_cast` necessary? I was under the impression they were just assignable

**Comment by VivaanBahl** [comment]
[GPSNode - Break out GPGGA parsing](https://trello.com/c/kAEjjXZl/112-gpsnode-break-out-gpgga-parsing)

**Comment by pranavmk98** [comment]
Made the required modifications - didn't change the static_cast<float> for the utm_point.northing and easting to be safe.

File: Software/real_time/ROS_RoboBuggy/src/robobuggy/test/gps/GPS_LatLon_Test.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+{
+    Transistor_GPS_Broadcaster broadcaster;
+    broadcaster.convert_to_longitude("");
+}
+
+TEST(GPS_Test_Suite, gps_test_gpgga_valid)
+{
+    Transistor_GPS_Broadcaster broadcaster;
+    std::string valid[] = {"$GPGGA", "i am not an int but function should not care"};
+    EXPECT_TRUE(broadcaster.gpgga_valid(valid));
```
Comment (VivaanBahl) [inline Software/real_time/ROS_RoboBuggy/src/robobuggy/test/gps/GPS_LatLon_Test.cpp]: We should change all the `EXPECT`s to `ASSERT`s

**Comment by VivaanBahl** [comment]
abandoning in favor of new GPS redesign

**Pull request closed without merging**